pyquil/_parser/parser.py

```diff
@@ -142,10 +143,7 @@ def def_frame(self, frame, *specs):
         for (spec_name, spec_value) in specs:
             name = names.get(spec_name, None)
             if name:
-                if isinstance(spec_value, str) and spec_value[0] == '"' and spec_value[-1] == '"':
-                    # Strip quotes if necessary
-                    spec_value = spec_value[1:-1]
-                options[name] = spec_value
+                options[name] = json.loads(str(spec_value))
```

Comment:
Were `int` or `float` legal values here previously? If so, they'll be converted to strings now (and surrounded with quotes later) when they previously wouldn't have.

Reply:
Not true, and covered by the new test in this MR.

I also just double checked in the REPL in Python 3.7 and 3.9 just to be sure:
```python
>>> import json
>>> type(json.loads(str(3)))
<class 'int'>
>>> type(json.loads(str(3.0)))
<class 'float'>
>>> type(json.loads(str("false")))
<class 'bool'>
>>> type(json.loads(str("null")))
<class 'NoneType'>
```

Reply:
Oh, awesome! I had something backwards in my head, sorry for the false alarm 😓

Reply:
Np, thanks for reviewing!